mypy/config_parser.py

```diff
@@ -281,6 +417,17 @@ def parse_section(prefix: str, template: Options,
     return results, report_dirs
 
 
+def convert_to_boolean(value: Optional[Any]) -> bool:
+    """Return a boolean value translating from other types if necessary."""
+    if isinstance(value, bool):
+        return value
+    if not isinstance(value, str):
+        value = str(value)
+    if value.lower() not in configparser.RawConfigParser.BOOLEAN_STATES:
+        raise ValueError('Not a boolean: %s' % value)
```

Comment:
Similar to above, raise a custom error and make sure we handle it without generating a traceback.

Reply:
In this case, this is the same way that the `RawConfigParser` does it, so I'd recommend staying consistent with how it parses booleans vs. how we do so for the TOML file.

I.e., here's what's in `configparser.py`:

```python
    def _convert_to_boolean(self, value):
        """Return a boolean value translating from other types if necessary.
        """
        if value.lower() not in self.BOOLEAN_STATES:
            raise ValueError('Not a boolean: %s' % value)
        return self.BOOLEAN_STATES[value.lower()]
```